tlv/complex.go

```diff
@@ -0,0 +1,48 @@
+package tlv
+
+import (
+	"io"
+)
+
+// IndexStatus is an enum-like type that describes what state the indexed
+// outpoint is in. Currently only two possible values.
+type IndexStatus uint8
```

Comment:
typically this sort of type wouldn't be defined in the `tlv` package, but rather in `channeldb` in this case.